src/timeAgoInWords.README.md

```diff
@@ -0,0 +1,39 @@
+
+# `timeAgoInWords`
+
+Takes an absolute date in [epoch format](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Date#description) and returns a string representing relative time ago.
+
+## Usage
+`timeAgoInWords(epoch, opts)`
+
+Returns: `string | false`
+
+Converts an absolute epoch to a relative time ago string
+
+`epoch`
+
+Type: `number`
+
+The date when an event happened in epoch format
+
+`opts`
+
+Type:
+```typescript
+{
+	maxAge?: number; // Return false if the date ago is past a cutoff
+	format?: 'short' | 'med' | 'long'; // Change string format (default 'med')
+	showTime?: boolean; // Append time to string
+}
+```
+
+Options to control the response
+
+## Examples
+```
+timeAgoInWords(twoDaysAgoAsEpoch) // '2d ago'
+timeAgoInWords(twoDaysAgoAsEpoch, { format: 'long' }) // '2 days ago'
+timeAgoInWords(twoDaysAgoAsEpoch, { format: 'short' }) // '2d'
+timeAgoInWords(threeYearsAgo, { maxAge: 1YearAgo }) // false
+timeAgoInWords(threeDaysAgo, { showTime: true, format: 'long' }) // 'Friday 12 March 2021 10:00'
```

Comment:
I’d much prefer `March`, but according to the current code, this is not consistent 😉 

```suggestion
timeAgoInWords(threeDaysAgo, { showTime: true, format: 'long' }) // 'Friday 12 Mar 2021 10:00'
```

Reply:
I think this is interesting https://wmich.edu/writing/rules/dates#:~:text=March%2C%20April%2C%20May%2C%20June,.%2C%20Nov.%2C%20Dec. obviously, it would be more ideal to follow our own style guide but there is no reference on where to abbreviate and where not to: 
https://www.theguardian.com/guardian-observer-style-guide-d#sign-in-gate:~:text=dates,sixteen%E2%80%9D%2C%20not%20%E2%80%9Ctwo%20thousand%20and%20sixteen%E2%80%9D

This feels like we should ask Central Prod?

Reply:
I don't see anywhere we're using `long` so it's hard to know the uuse case but I'm inclined to agree that `long` means 'March'

Reply:
The api is confusing in this area and consequently so is the README. `long` doesn't mean a long absolute date, it means the relative date should be longer

```
short = 1s
med = 1s ago
long = 1 second ago
```

The confusion starts when dates pass 5 and 7 days. We loading more meaning into `long` in this case and returning absolute dates earlier. So with `med` we would return `6d ago` but with `long` this would become absolute.

'Friday 12 Mar 2021' is how all absolute dates look (not just 'long' dates) so any decision should be considered with this in mind

Reply:
Maybe we could simplify this area by removing this special logic? The `long` option is never used, not in Frontend, nor in DCR, so I don't think we need this code:

```
        const extendedFormatting = opts.format === 'short' || opts.format === 'med';

        ... if (isToday(then) || (extendedFormatting && isWithin24Hours(then))) {
		// Hours
		...
	} else if (extendedFormatting && isWithinPastWeek(then)) {
		// Days
		...
	} else  if (isYesterday(then)) {...
```

You could refactor that to:

```
        ... if (isWithin24Hours(then)) {
		// Hours
		...
	} else if (isYesterday(then)) {
		// Yesterday
		...
	} else if (isWithinPastWeek(then)) {
		// Days
		...
	} else ...
```

and get the same result and then `long` just means the relative string is longer, which is still never used but nice to have as an option.

Reply:
> 'Friday 12 Mar 2021' is how all absolute dates look

is this an unexpected result for a method called `timeAgoInWords`? especially is nothing is using it

how about default is medium, with an optional `compact: Boolean` param that returns short, and remove long since it's actually an absolute. 

_or..._

if this is _really_ a guardian date formatter, options something like `compact: Boolean` and `relative: Boolean` so that:

```js
dateTimeInWords(epoch) // 'Friday 12 March 2021 10:00' 
dateTimeInWords(epoch, { compact: true }) // 'Friday 12 Mar 2021 10:00'
dateTimeInWords(epoch, { relative: true }) // '2 days ago'
dateTimeInWords(epoch, { relative: true, compact: true }) // '2d'
```

Reply:
actually, maybe this:
```js
dateTimeInWords(epoch) // 'Friday 12 Mar 2021 10:00' 
```
and 
```js
timeAgoInWords(epoch) // '2 days ago'
timeAgoInWords(epoch, { compact: true }) // '2d'
```

because the point of something like `dateTimeInWords` would be to get a consistent format, and it's already settled on the short version

Reply:
You're right, `timeAgoInWords` isn't being truthful! 🤔 

So the logic, simplified, is:

```
	if (secondsAgo < 0) {
		// Dates in the future are not supported
	} else if (within55Seconds) {
		// Seconds
	} else if (withinTheHour) {
		// Minutes
	} else if (within24hrs) {
		// Hours
	} else if (wasYesterday) {
		// Yesterday
	} else if (within5Days) {
		// Days
	} else if (withinTheWeek) {
		// Include day of week in string - "Friday 15 Nov 2019 13:00"
	}
	return (
		// Over a week, return a simple date - "9 Nov 2019"
	);
```

+1 for using a simpler api (always) but for DCR to keep parity we'd need it to be:

```
dateTimeInWords(epoch) // '2d ago'
dateTimeInWords(epoch, { compact: true }) // '2d'
```

Thus, losing `2 days ago` as an option. Maybe this is fine? We're not using it anywhere so there's no immediate impact, it just means it's not available as a future format (at least not without a bit of work)

Reply:
Arguably, `showTime` could be removed as well? Again, it's not used anywhere outside of the tests.

Reply:
It seems a little curious to have the day of the week only for day 6 or 7. You wrote earlier that all absolute dates contain the day of the week, but the sample code hints otherwise.

Reply:
I'm working backwards from the orginal code which has some weird edge cases around combinations of length ago + format === long. I'm guess this was battlescar logic left over from living in the wild for a while. I think the intention was to be more descriptive when days of week might be useful and then to fall back to pure dates afterwards?

I'd be in favour of simplifying further as this will keep the code easier to understand and the output shorter?

Reply:
i didn't mean to leave `dateTimeInWords` in, i'm really sorry, that was a hangover from an idea i dismissed – but it does help clear up the absolute thing i think, and @mxdvl's comment speaks to the same issue, which is that after a time it's useful to talk about absolutes and not relative dates. and that time might be different in different places.

how about adding `daysUntilAbsolute: Number = 7`?

also,

>Thus, losing 2 days ago as an option.

begs the question, why return it? if we never returned it then it would be up to the user to add it if useful.

so then:

```js
// epoch is 2 days ago
timeAgo(epoch) // '2 days'
timeAgo(epoch, { compact: true }) // '2d'

// epoch is 10 days ago
timeAgo(epoch) // '9 November 2019'
timeAgo(epoch, { compact: true }) // '9 Nov 2019'
timeAgo(epoch, { daysUntilAbsolute: 30 }) // '10 days'
timeAgo(epoch, { compact: true, daysUntilAbsolute: 30 }) // '10d'
```

Reply:
😍 `timeAgo` 😍 Brilliant!

Thanks for this suggestion and the other api improvements - really great input! I've implenented the api described above with two exceptions:

1. I kept 'ago'. Both DCR and Frontend expect this by default and I was concerned removing it could lead to '9 Mar 2021 ago` type bugs. I think it's nice to keep the decision about if ago is needed or not in here. 

Sorry, I think my comment above around 'ago'  might have been confusing.

2. I switched `compact` to `extended`. I prefer the _sound_ of `compact`, it reads nice, but the default for Frontend and DCR is to be compact and the longer, extended, format is the optional branch here. So it made more sense to default to compact when no option is given and only use the `extended` format in special situations. Er, if that makes sense!? Basically, all the existing code using this function (or the local version of it) defaults to `2d ago` and so in order to have a drop in replacement, we need `extended` to be the option.

Reply:
> '9 Mar 2021 ago`

oh yeah of course! nice one

Reply:
> it made more sense to default to compact when no option is given and only use the extended format in special situations

👍

Reply:
how do you feel about `verbose` over `extended`? don't want to keep this running forever, so up to you, not a blocker

Reply:
Ha, so I actually searched the internet for 'compact antonym' and got this
![Screenshot 2021-03-18 at 17 20 39](https://user-images.githubusercontent.com/1336821/111669039-4c8a8c00-880e-11eb-9052-6fc1e6a89b8b.jpg)
So that didn't go well.

I think your idea of 'verbose', which means wordy, is better than 'extended', which implies adding to something. So 👍 

And I'm happy to let this run. Loving the review!

Reply:
`extended` has been replaced with `verbose` 👍